### DISCUSSION TITLE
Programming new/unsupported devices

### DISCUSSION THREAD

**Discussion by jobsoftinc** [Koenkk/zigbee2mqtt#6803] [Devices]
I've read the docs, but am still not 100% clear on how one fully integrates a new or unsupported device. I have some Sengled PAR38 motion sensing bulbs (E13-N11) that I got to work by cloning the regular A19/E26 Sengled bulb "module" in Z2M. They work, but only as a bulb. How do I determine what else the E13-N11's can do (attribute/parameter wise), and then what's the best way to incorporate that into the cloned Z2M module? These are motion sensing bulbs, and there is also a parameter to turn on/off the self "on" (with motion detection) function (when using a Sengled hub, of course). Should I have setup a sniffer node to capture what was going on between the bulbs and the sengled hub? These are fairly nice bulbs for the price. And it would be great to have all their features incorporated into Z2M. I'm just not sure what's the best workflow approach to go about it. Thanks!

**Comment by Faustino-ai** [comment]
I am also quite new but I would love to see support for such a device. I will try to help as much as I could. From my own personal understanding (at this stage is still quite poor), if you do not receive any message for "No converter available for......" then your device probably needs to have it's traffic sniffed to be able to move forward. The workflow should be something like:
1. Checking the log from zigbee2mqtt with debug on to see if there's any converter missing. If there is, follow the guide until no converter is missing. Then proceed to the next step.
2. Sniff the traffic when you use the device for the missing functions that you get if you use the out of the box setup (the native app + original zigbee gateway)
3. Decrypt the payloads and try to implement them. I cant help much on this stage because I personally am stuck at this stage.
